### PR TITLE
Switch from DllExport to DNNE in SosThreadingTools

### PR DESCRIPTION
**Pull request by AArnott**

### CODE REVIEW

File: src/SosThreadingTools/SosThreadingTools.csproj [hunk 1]
```diff
@@ -1,15 +1,13 @@
 ﻿<Project Sdk="MSBuild.Sdk.Extras">
   <PropertyGroup>
-    <TargetFrameworks>net472</TargetFrameworks>
+    <TargetFrameworks>net7.0-windows</TargetFrameworks>
     <RuntimeIdentifiers>win-x86;win-x64</RuntimeIdentifiers>
     <ExtrasBuildEachRuntimeIdentifier>true</ExtrasBuildEachRuntimeIdentifier>
+    <EnableDynamicLoading>true</EnableDynamicLoading>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
 
     <AssemblyName>$(MSBuildProjectName)</AssemblyName>
```
Comment (AaronRobinsonMSFT) [inline src/SosThreadingTools/SosThreadingTools.csproj:9]: You might want to change this to `<AssemblyName>$(MSBuildProjectName)Impl</AssemblyName>` and then create `<DnneNativeBinaryName>$(MSBuildProjectName)</DnneNativeBinaryName>` for the native binary.

Comment (AArnott) [reply src/SosThreadingTools/SosThreadingTools.csproj:9]: Agreed.

File: src/SosThreadingTools/SosThreadingTools.csproj [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
+    <TargetFrameworks>net7.0-windows</TargetFrameworks>
     <RuntimeIdentifiers>win-x86;win-x64</RuntimeIdentifiers>
     <ExtrasBuildEachRuntimeIdentifier>true</ExtrasBuildEachRuntimeIdentifier>
+    <EnableDynamicLoading>true</EnableDynamicLoading>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
 
-    <AssemblyName>$(MSBuildProjectName)</AssemblyName>
-    <DebugSymbols>true</DebugSymbols>
-    <DllExportOurILAsm>false</DllExportOurILAsm>
+    <!-- This project's build is incomplete without DNNE, which only works on Windows. -->
```
Comment (AaronRobinsonMSFT) [inline src/SosThreadingTools/SosThreadingTools.csproj:9]: This isn't true. DNNE works on all platforms but uses the TFM to understand what platform to target. This is a fundamental difference between unmanaged and managed code and nothing to do with DNNE. Setting the TFM to net7.0 would permit this to build.

Comment (AArnott) [reply src/SosThreadingTools/SosThreadingTools.csproj]: I've modified the comment to specify that DNNE only works on Windows because of our -windows TFM.

**Review by AaronRobinsonMSFT — APPROVED** [review]
The usage of DNNE looks good to me.

**Pull request merged**